src/Components/HomePage.js

```diff
@@ -0,0 +1,200 @@
+import { NavLink } from 'react-router-dom';
+import { MdSettings } from 'react-icons/md';
+import { FaMicrophone } from 'react-icons/fa';
+import { BsArrowRightCircle } from 'react-icons/bs';
+import {
+  getAllCoins, getCoinMarkets, searchByRank, searchBycrypto,
+} from '../Redux/CoinAPI';
+import cryptoPix from './images/crypto-pix.jpg';
+
+const HomePage = () => {
+  const getList = useSelector((state) => state.cryptoCoins);
+
+  const coins = getList.coin;
+  const { currency } = getList;
+
+  const dispatch = useDispatch();
+
+  let cur = 'USD';
+
+  const onSelect = (e) => {
+    cur = e.target.value;
+    dispatch(getAllCoins(cur));
+    const parent = e.target.parentElement.children[1];
+    parent.children[0].value = '';
+  };
+
+  useEffect(() => {
+    dispatch(getAllCoins(cur));
+  }, []);
+
+  let rank = '';
+
+  const onFillValue = (e) => {
+    rank = e.target.value;
+  };
+
+  const onSearch = (e) => {
+    e.preventDefault();
+
+    if (rank) {
+      dispatch(searchByRank(rank));
+    } else if (!rank) {
+      dispatch(getAllCoins(cur));
+    }
+  };
+
+  const onClear = (e) => {
+    dispatch(getAllCoins(cur));
+    e.target.parentElement.firstElementChild.value = '';
+  };
+
+  const onBackspace = (e) => {
+    if (e.key === 'Backspace') {
+      dispatch(getAllCoins(cur));
+    }
+  };
+
+  const handleClick = (e) => {
+    dispatch(getCoinMarkets(e.target.id));
+  };
+
+  let searchCurrency = '';
+
+  const getSerachValue = (e) => {
+    searchCurrency = e.target.value;
+  };
+
+  const onSearchCurrency = (e) => {
+    e.preventDefault();
+    if (searchCurrency) {
+      dispatch(searchBycrypto(searchCurrency));
+    } else if (!searchCurrency) {
+      dispatch(getAllCoins(cur));
+    }
+  };
+
+  return (
+    <div className="pg-container">
+      <nav>
+        <p className="heading">Top Cryptos</p>
+        <div className="nav-icon">
+          <FaMicrophone style={{ color: 'white', marginRight: '2rem' }} />
+          <MdSettings style={{ color: 'white' }} />
+        </div>
+      </nav>
+      <div className="pg-head">
+        <div className="crypto-pix">
+          <img className="image" src={cryptoPix} alt="currency" />
+        </div>
+        <div className="head-desr">
+          <h1 className="crypto-x">CRYPTO EXCHANGE</h1>
+        </div>
+
+      </div>
+
+      <div className="coin-heading">
+        <div>
+          <h4>Price Per Coin</h4>
+
+          {/* <form onSubmit={(e) => onSearchCurrency(e)}>
+            <input
+              name="search"
+              type="text"
+              className="search-field"
+              placeholder="Search By Currency"
+              onChange={(e)=> getSerachValue(e)}
+              onKeyDown={(e) => onBackspace(e)}
+            />
+            <button className="filter-btn" type="submit">Search</button>
+            <button className="clear-btn" type="button" onClick={(e) => onClear(e)}>Clear</button>
+          </form> */}
```

Comment:
- [ ] [Optional] Kindly remove commented codes from your code base, for a cleaner code base